sw/airborne/firmwares/fixedwing/stabilization/stabilization_attitude.c

```diff
@@ -472,9 +473,8 @@ inline static void h_ctl_pitch_loop(void)
 #endif
 
 
-  float d_err = err - last_err;
-  last_err = err;
-  float cmd = -h_ctl_pitch_pgain * (err + h_ctl_pitch_dgain * d_err);
+
+  float cmd = -1*(h_ctl_pitch_pgain * err + h_ctl_pitch_dgain * body_rate->q);
```

Comment:
This is actually breaking all airframe files using the legacy control loops. Before the pgain was in factor of the angle error and the derivative term, as described here: http://wiki.paparazziuav.org/wiki/Control_Loops#Pitch_loop.
With your modification, it makes all previous pitch damping factor ineffective: pgain is usually in the range of a few thousands (since `cmd` is between [-9600, 9600]) and dgain around 1).
The reason to have this, is that in this form the dgain represent the time constant associated with the derivator, and is independent of the overall gain (pgain here). As a side note, the "new/adaptive" control is using the other usual form kp*e + kd*de/dt as you suggest here.
So finally, I'm not really in favor of changing the control structure now.

Regarding the choice of a soft derivate instead of using pitch rate, it is just that this piece of code is older than the generalized usage of IMU and hence gyro data on the pitch axis.
This is how it is handled in the adaptive loop: https://github.com/paparazzi/paparazzi/blob/master/sw/airborne/firmwares/fixedwing/stabilization/stabilization_adaptive.c#L600 with a USE_GYRO_PITCH_RATE default to true.